src/NuGet.Core/NuGet.Protocol/LegacyFeed/V2FeedParser.cs

```diff
@@ -503,44 +503,65 @@ private static string GetString(XElement parent, XName childName)
             ILogger log,
             CancellationToken token)
         {
-            return await _httpSource.ProcessResponseAsync(
-                new HttpSourceRequest(
-                    () =>
-                    {
-                        var request = HttpRequestMessageFactory.Create(HttpMethod.Get, uri, log);
-                        request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/atom+xml"));
-                        request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/xml"));
-                        return request;
-                    }),
-                async response =>
+            // Try the call up to 3 times.
+            for (var retry = 0; retry < 3; ++retry)
+            {
+                try
                 {
-                    if (response.StatusCode == HttpStatusCode.OK)
-                    {
-                        var networkStream = await response.Content.ReadAsStreamAsync();
-                        return await LoadXmlAsync(networkStream);
-                    }
-                    else if (ignoreNotFounds && response.StatusCode == HttpStatusCode.NotFound)
-                    {
-                        // Treat "404 Not Found" as an empty response.
-                        return null;
-                    }
-                    else if (response.StatusCode == HttpStatusCode.NoContent)
-                    {
-                        // Always treat "204 No Content" as exactly that.
-                        return null;
-                    }
-                    else
-                    {
-                        throw new FatalProtocolException(string.Format(
-                            CultureInfo.CurrentCulture,
-                            Strings.Log_FailedToFetchV2Feed,
-                            uri,
-                            (int)response.StatusCode,
-                            response.ReasonPhrase));
-                    }
-                },
-                log,
-                token);
+                    return await _httpSource.ProcessResponseAsync(
+                        request: new HttpSourceRequest(() => CreateRequest(uri, log)),
+                        processAsync: response => ProcessResponseAsync(response, uri, ignoreNotFounds, token),
+                        log: log,
+                        token: token);
+                }
+                catch (Exception ex) when (retry < 2)
```

Comment:
Still insisting on narrow subset of exceptions including `FatalProtocolException` and the exceptions thrown by functional tests. The change should be less intrusive while targeting failing tests first as these tests represent actual network issues NuGet user may encounter. We definitely don't want to retry on `OutOfMemoryException` or `StackOverflowException`.